src/core/route/util/routeUtils.ts

```diff
@@ -0,0 +1,11 @@
+function generateRedirectStateFromLocation(
+  location: any,
```

Comment:
We can use `LocationDescriptorObject` type from `history`

```suggestion
  location: LocationDescriptorObject,
```